### DISCUSSION TITLE
Project files deleted when I pull the code from the repository

### DISCUSSION THREAD

**Discussion by Divyanshu-prog237** [community/community#121091] [Repositories]
I push and pull the code from the repository, accidentally my project files deleted or vanished out now I'm unable to access it, what I do

**Comment by ljste** [comment]
Hey Divyanshu-prog237,

Sorry to hear about the trouble with your project files! Here’s what you can do to recover your files and prevent this from happening again:

Check Your Local Backups: First, see if you have any local backups of your files. If you're using a version control system, there might be local copies that haven't been affected.

Recover Deleted Files: If you're on Windows, check the Recycle Bin to see if the files are there. For macOS, look in the Trash. You can also use file recovery software to try to recover deleted files from your disk.

Check Repository History: Use your version control system's log or history command (like git log or svn history) to check for the last commits where the files were still present. You can potentially revert to those commits to restore your files.

Prevent Future Issues: To avoid similar problems in the future, consider setting up more frequent commits and backups. Also, always make sure to double-check which files are being affected before running commands that modify your repository (like git pull or git push).

Hope this helps you get back on track!
